tests/test_checkers.py

```diff
@@ -995,6 +998,59 @@ def test_text_real_file(self, tmp_path: Path):
             check_type(f, TextIO)
 
 
+class TestIntersectingProtocol:
+    SIT = TypeVar("SIT", bound=object, covariant=True)
+
+    class SizedIterable(
+        Sized,
+        Iterable[SIT],
+        Protocol[SIT],
+    ): ...
+
+    @pytest.mark.parametrize(
+        ("subject", "predicate_type"),
+        (
+            ((), SizedIterable),
+            (range(2), SizedIterable),
+            ((), SizedIterable[int]),
+            ((1, 2, 3), SizedIterable[int]),
+            (("1", "2", "3"), SizedIterable[str]),
```

Comment:
Add `pytest.param() and the `id=` parameter to distinguish between individual test cases.